### ISSUE TITLE
Make torch.logsumexp operate over dimension 0 by default (or at least for 1D arrays)

### ISSUE THREAD

**Issue by chausies** [pytorch/pytorch#19149]
It's pretty ridiculous that, for a 1D input like `x = torch.rand(5)`, `torch.logsumexp` still requires you to specify you're operating over dimension 0.

In particular `torch.logsumexp(x)` will error, saying that it needs a dim argument. `torch.logsumexp(x, 0)` works.

My suggestion is that, at the very least for 1D inputs, the default dim you lse over is set to 0.

**Comment by soumith** [comment]
this is fair, we should do it.

**Comment by zhangguanheng66** [comment]
We should remember to update our documentation for this change.

**Comment by chausies** [comment]
Any thoughts on, for non-1D tensors, what the default dimension should be (if there is one)? There's arguments for 0 and -1 as the defaults. For the record, numpy and scipy default to `None`, which indicate operating over every last entry in every last dimension. For `cumsum`, this means operating over the flattened array.

**Comment by gchanan** [comment]
NumPy's behavior seems reasonable.
